documentation/swagger-extensions.md

```diff
@@ -531,6 +532,94 @@ parameters | [Array of Parameter Objects](https://github.com/OAI/OpenAPI-Specifi
+- When a property is modeled as `"readonly": true` then,
+  - if the x-ms-mutability extension is applied then it can **only have "read" value in the array**. 
+  - applying the extension as `"x-ms-mutability": ["read"]` or not applying it will have the same effect.
+- When the property is modeled as **`"readonly": false`** then,
+  - applying the extension as `"x-ms-mutability": ["read"]` is not allowed.
+  - applying the extension as `"x-ms-mutability": ["create", "read", "update"]` or not applying it will have the same effect.
+  - applying the extension with anyother **permissible valid combination** should be fine.
+- When this extension is applied on a collection (array, dictionary) then this will have effects on the mutability (adding/removing elements) of the collection. Mutabiility of the collection cannot be applied on its elements. The mutability of the element will be governed based on the mutability defined in the element's definition.
+
+Examples:
+- Mutability on a model definition
+```json5
+"definitions": {
+  "Resource": {
+    "description": "The Resource Model definition.",
+    "properties": {
+      "id": {
+        "readOnly": true,
+        "type": "string",
+        "description": "Resource Id",
+        "x-ms-mutability": ["read"]
+      },
+      "name": {
+        "type": "string",
+        "description": "Resource name"
+      },
+      "type": {
+        "type": "string",
+        "description": "Resource type",
+        "x-ms-mutability": ["read"]
+      },
+      "location": {
+        "type": "string",
+        "description": "Resource location",
+        "x-ms-mutability": ["create", "read"]
+      },
+      "tags": {
+        "type": "object",
+        "additionalProperties": {
+          "type": "string"
+        },
+        "description": "Resource tags",
+        "x-ms-mutability": ["create", "read", "update"]
+      }
+    },
+    "required": [
+      "location"
+    ],
+    "x-ms-azure-resource": true
+  }
+}
+```
+- Mutability of the object property; which is a collection of items
+```json5
+"definitions": {
+  "ResounceCollection": {
+    "description": "Collection of Resource objects. Resource is defined in the above example.",
+    "properties": {
+      "value": {
+        "type": "array",
+        "description": "Array of Resource objects.",
+        "x-ms-mutability": ["create", "read", "update"], //This means that the array is mutable
+        "items": {
+          "type": object,
+          "x-ms-mutability": ["create", "read"] // X - Applying mutability on the itemType of the array or vauleType of the dictionary is not allowed.
```

Comment:
I would use a stronger notation than "X" to indicate that something is not allowed...